packages/turf-flatten/index.ts

```diff
@@ -19,10 +30,24 @@ import { featureCollection } from "@turf/helpers";
  * //addToMap
  * var addToMap = [flatten]
  */
-function flatten(geojson) {
+function flatten<T extends Point | MultiPoint>(
+  geojson: Feature<T> | FeatureCollection<T> | T
+): FeatureCollection<Point>;
+
+function flatten<T extends LineString | MultiLineString>(
+  geojson: Feature<T> | FeatureCollection<T> | T
+): FeatureCollection<LineString>;
+
+function flatten<T extends Polygon | MultiPolygon>(
+  geojson: Feature<T> | FeatureCollection<T> | T
+): FeatureCollection<Polygon>;
+
+function flatten(geojson: AllGeoJSON): FeatureCollection<any>;
+
+function flatten(geojson: AllGeoJSON): FeatureCollection {
```

Comment:
Personally I find the nested ternaries pretty difficult to parse, so the overloaded function definitions are more readable. Possibly also what the casual consumer would prefer?

Reply:
Yeah I maybe buy that argument. The ternaries aren't too bad if they're formatted appropriately it winds up like a big if/else, but its maybe harder to know how to read them as a consumer if you're getting some inferred type instead of something concrete.